pyscf/neo/tdgrad.py

```diff
@@ -0,0 +1,858 @@
+            veff0doo = vj[0] * 2 - vk[0] + f1oo[0] + k1ao[0] * 2
+            wvo_e = reduce(numpy.dot, (orbv['e'].T, veff0doo, orbo['e'])) * 2
+            if singlet:
+                veff = vj[1] * 2 - vk[1] + f1vo[0] * 2
+            else:
+                veff = f1vo[0] - vk[1]
+            veff0mop = reduce(numpy.dot, (mo_coeff['e'].T, veff, mo_coeff['e']))
+            wvo_e -= numpy.einsum('ki,ai->ak', veff0mop[:nocc['e'],:nocc['e']], xpy) * 2
+            wvo_e += numpy.einsum('ac,ai->ci', veff0mop[nocc['e']:,nocc['e']:], xpy) * 2
+            veff = -vk[2]
+            veff0mom = reduce(numpy.dot, (mo_coeff['e'].T, veff, mo_coeff['e']))
+            wvo_e -= numpy.einsum('ki,ai->ak', veff0mom[:nocc['e'],:nocc['e']], xmy) * 2
+            wvo_e += numpy.einsum('ac,ai->ci', veff0mom[nocc['e']:,nocc['e']:], xmy) * 2
+        else:
+            vj = mf_e.get_j(mol_e, (dmzoo, dmxpy+dmxpy.T), hermi=1)
+            veff0doo = vj[0] * 2 + f1oo[0] + k1ao[0] * 2
+            wvo_e = reduce(numpy.dot, (orbv['e'].T, veff0doo, orbo['e'])) * 2
+            if singlet:
+                veff = vj[1] * 2 + f1vo[0] * 2
+            else:
+                veff = f1vo[0]
+            veff0mop = reduce(numpy.dot, (mo_coeff['e'].T, veff, mo_coeff['e']))
+            wvo_e -= numpy.einsum('ki,ai->ak', veff0mop[:nocc['e'],:nocc['e']], xpy) * 2
+            wvo_e += numpy.einsum('ac,ai->ci', veff0mop[nocc['e']:,nocc['e']:], xpy) * 2
+            veff0mom = numpy.zeros((nmo_e,nmo_e))
+
+    wvo = {}
+    wvo['e'] = wvo_e
+    dm = {}
+    for t in mf.components.keys():
+        dm[t] = None
+    dm['e'] = dmzoo
+    for t_pair, interaction in mf.interactions.items():
+        p1, p2 = t_pair
+        if p1.startswith('e'):
+            vj_en = interaction.get_vint(dm)
+            wvo_n = reduce(numpy.dot, (orbv[p2].T, vj_en[p2], orbo[p2])) * 2
+            wvo[p2] = wvo_n
+
+    vresp = mf.gen_response(max_memory=max_memory, hermi=0)
+    def fvind(mo1, f1):
+        dm = {}
+        for t in mo1.keys():
+            mo1[t] = mo1[t].reshape(nvir[t], nocc[t])
+            dm_t = reduce(numpy.dot, (orbv[t], mo1[t], orbo[t].T))
+            dm[t] = dm_t + dm_t.T
+        v1ao = vresp(dm)
+        v1ao['e'] *= 2
+        for t in f1.keys():
+            f1[t] = f1[t].ravel()
+        f_add = get_fock_add_cdft(f1, int1e_r_ao)
+
+        v1 = {}
+        for t in v1ao.keys():
+            if t.startswith('n'):
+                v1ao[t] += f_add[t]
+            v1[t] = reduce(numpy.dot, (orbv[t].T, v1ao[t], orbo[t])).ravel()
+
+        rfn = position_analysis(mo1, int1e_r_vo)
+
+        return v1, rfn
+
+    z1, mo_e1, f1 = cphf.solve_nos1(fvind, mf.mo_energy, mo_occ, wvo,
+                                          with_f1 = True,
+                                          max_cycle=td_grad.cphf_max_cycle,
+                                          tol=td_grad.cphf_conv_tol,
+                                          verbose = verbose)
```

Comment:
Align with fvind. There are many other places that I did not comment about. Check PEP8 recommended styles, which even linter won't complain about, but are good to follow.